### PR TITLE
[PWD-2541] Finalize On Demand Contact Us Modal

### PR DESCRIPTION
**Pull request by FahemHakikiKhaya**
FIX: 
1. Reset selected topic when program change
2. Remove prefix coma if otherTopic is filled and normal topic isnot filled

image

### CODE REVIEW

File: src/components/ContactUs/OnDemand/formik-hooks/useFormikStep3.ts [hunk 1]
```diff
@@ -116,10 +116,14 @@ const useFormikStep3 = ({
...
       const topicValues = () => {
         if (otherTopicCheckbox) {
-          return topic.concat(',', otherTopic)
+          if (topic) {
+            return topic.concat(',', otherTopic)
+          } else {
+            return otherTopic
+          }
+        } else {
+          return topic
```
Comment (katherinedavenia) [inline src/components/ContactUs/OnDemand/formik-hooks/useFormikStep3.ts:125]:
```suggestion
    const topicValues = () => {
       if (otherTopicCheckbox && topic) {
           return topic.concat(',', otherTopic);
        }
       return topic || otherTopic;
    };
```
gini lebih enak gak sii

Comment (FahemHakikiKhaya) [reply src/components/ContactUs/OnDemand/formik-hooks/useFormikStep3.ts:125]: iya jugaa

File: src/components/ContactUs/OnDemand/formik-hooks/useFormikStep3.ts [hunk 2]
```diff
@@ -154,19 +153,20 @@ const useFormikStep3 = ({
       }
 
       // Handle corporate training newsletter subscription
-      if (subscribeNewsletterCheckbox) {
-        void createMailchimp(dataMailchimp)
-      }
+      console.log(topicValues())
```
Comment (Vesals) [inline src/components/ContactUs/OnDemand/formik-hooks/useFormikStep3.ts:156]: log hem

File: src/components/ContactUs/OnDemand/formik-hooks/useFormikStep3.ts [hunk 3]
```diff
@@ -154,19 +153,20 @@ const useFormikStep3 = ({
...
-      setToggleModalSuccess(true)
+      // onClose()
+      // setToggleModalSuccess(true)
 
-      resetFormStep1()
-      resetFormStep2()
-      resetFormStep3()
+      // resetFormStep1()
+      // resetFormStep2()
+      // resetFormStep3()
```
Comment (shadiqfauzi) [inline src/components/ContactUs/OnDemand/formik-hooks/useFormikStep3.ts:169]: kenapa ini malah jadi ke comment brow

**Review by katherinedavenia — APPROVED** [review]

**Review by haikal1223 — APPROVED** [review]

**Pull request merged**